presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java

```diff
@@ -327,21 +328,32 @@ private QueryResult executeQueryTest()
             }
             while (retry < testTeardownRetries);
             // if teardown is not successful the query fails
-            queryResult = tearDownResult.getState() == State.SUCCESS ? queryResult : tearDownResult;
+            if (tearDownResult.getState() != State.SUCCESS) {
```

Comment:
@nezihyigitbasi I think the logic should be different. If setup failed, then it should be the main failure. The teardown failure, if present, should be added to setup failure. Similarly, if query failed, then it should be the main failure.